portfolio/src/main/webapp/script.js

```diff
@@ -28,11 +28,23 @@ function addRandomGreeting() {
   greetingContainer.innerText = greeting;
 }
 
-function getHelloMessage(){
+// Fetches a JSON with messages from the server, creates elements with the messages and adds them to the DOM:
+function getMessages(){
 
-    fetch("/data").then(response => response.text()).then((HelloMessage) => {
+    fetch("/data").then(response => response.json()).then((messages) => {
 
-        document.getElementById("message-zone").innerHTML = HelloMessage;
+        //Iterates the JSON by keys:
+        for(var key in messages){
```

Comment:
In this case it may be simper to just use a `for of` loop:

https://alligator.io/js/for-of-for-in-loops/

i.e.

```
for (let message of messages) {
  ...
  messageElement.innerText = message;
  ...
}
```

Reply:
Cool doc! I get it now. I should use `for of` in this case because after the parsing the JSON has no explicit keys (it would be the indexes) so its an array (iterable --> use for of).